### PR TITLE
ArC - improve the error message when a programmatic lookup problem occurs

### LINKED ISSUE #21218: Enhance warning about programmatic lookup problem detected with source of the problem
### Description

I am currently getting an Error about `programmatic lookup problem detected`.
However, since this project uses modules from other contributors, I have no idea where the programmatic lookup might be happening.

Can this warning please be enhanced with the source of the lookup?

```
================================================================================
CDI: programmatic lookup problem detected
-----------------------------------------
At least one bean matched the required type and qualifiers but was marked as unused and removed during build
Removed beans:
- CLASS bean  [types=[class com.acme.SomeBean], qualifiers=null]
Required type: class com.acme.SomeBean
Required qualifiers: [@javax.enterprise.inject.Default()]
Solutions:
- Application developers can eliminate false positives via the @Unremovable annotation
- Extensions can eliminate false positives via build items, e.g. using the UnremovableBeanBuildItem
- See also  -' class='text-primary' target='_blank'>https://quarkus.io/guides/cdi-reference#remove_unused_beans
- Enable the DEBUG log level to see the full stack trace to identify the method that performed the lookup
================================================================================
```

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #21218

### CODE REVIEW

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java [hunk 1]
```diff
@@ -630,22 +630,28 @@ private static Integer getAlternativePriority(InjectableBean<?> bean) {
...
                         + "\t- Extensions can eliminate false positives via build items, e.g. using the UnremovableBeanBuildItem\n"
                         + "\t- See also https://quarkus.io/guides/cdi-reference#remove_unused_beans\n"
-                        + "\t- Enable the DEBUG log level to see the full stack trace to identify the method that performed the lookup\n"
+                        + "\t- Enable the DEBUG log level to see the full stack trace\n"
                         + "%1$s%1$s%1$s%1$s\n";
+                StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
+                StackTraceElement element = Arrays.stream(stackTrace)
+                        .skip(1)
+                        .filter(ArcContainerImpl::isCallerElement)
+                        .findFirst().orElse(null);
```
Comment (geoand) [inline independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java:648]: Hm... I think the Stackwalker API would be very useful here

Comment (mkouba) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java:648]: Ah, didn't know about this one. Interesting. Although for this particular use case we don't need anything special...

Comment (geoand) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java:648]: Indeed, but I _think_ the StackWalker also perform better.

Comment (mkouba) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java]: Done.

**Review by geoand — APPROVED** [review]

**Review by manovotn — APPROVED** [review]

**Pull request merged**